src/os/bluestore/BlueStore.cc

```diff
@@ -11776,11 +11841,15 @@ int BlueStore::read(
     if (offset == length && offset == 0)
       length = o->onode.size;
 
-    r = _do_read(c, o, offset, length, bl, op_flags);
+    r = _do_read(c, o, offset, length, bl, op_flags, 0,
+          might_need_reformatting ? &span_stat : nullptr);
     if (r == -EIO) {
       logger->inc(l_bluestore_read_eio);
     }
   }
+  if (might_need_reformatting) {
+    _maybe_reformat_object(c, o, offset, length, bl, op_flags, span_stat);
+  }
```

Comment:
Now we will add latency of reformat to `l_bluestore_read_lat`. I am not sure whether we want it or not.